python/src/equistore/operations/multiply.py

```diff
@@ -34,10 +34,11 @@ def multiply(A: TensorMap, B: Union[float, TensorMap]) -> TensorMap:
 
     :return: New :py:class:`TensorMap` with the same metadata as ``A``.
     """
-
+    A = A.copy()
```

Comment:
I don't think this is necessary. The code below (_multiply_block_constant/_multiply_block_block) should not make any in-place modification, so this would just be wasting memory